big_tests/tests/sasl_external_SUITE.erl

```diff
@@ -85,6 +130,51 @@ cert_with_cn_no_xmpp_addrs_request_name_empty(C) ->
 
     escalus_connection:stop(Client).
 
+self_signed_cert_fails_to_authenticate_with_tls(C) ->
+    self_signed_cert_fails_to_authenticate(C, escalus_tcp).
+
+self_signed_cert_fails_to_authenticate_with_ws(C) ->
+    self_signed_cert_fails_to_authenticate(C, escalus_ws).
+
+self_signed_cert_fails_to_authenticate_with_bosh(C) ->
+    self_signed_cert_fails_to_authenticate(C, escalus_bosh).
+
+self_signed_cert_fails_to_authenticate(C, EscalusTransport) ->
+    Self = self(),
+    F = fun() ->
+		UserSpec = generate_user(C, "alice-self-signed", EscalusTransport),
+		{ok, Client, _} = escalus_connection:start(UserSpec),
+		Self ! escalus_connected,
+		escalus_connection:stop(Client)
+	end,
+    %% We spawn the process trying to connect because otherwise the testcase may crash
+    %% due linked process crash (client's process are started with start_link)
+    Pid = spawn(F),
+    MRef = erlang:monitor(process, Pid),
+
+    receive
+	{'DOWN', MRef, process, Pid, _Reason} ->
+	    ok;
+	escalus_connected ->
+	    ct:fail(authenticated_but_should_not)
+    after 10000 ->
+	      ct:fail(authenticated_but_should_not)
+    end.
+
+self_signed_cert_is_allowed_with_tls(C) ->
+    self_signed_cert_is_allowed_with(escalus_tcp, C).
+
+self_signed_cert_is_allowed_with_ws(C) ->
+    self_signed_cert_is_allowed_with(escalus_ws, C).
+
+self_signed_cert_is_allowed_with_bosh(C) ->
+    self_signed_cert_is_allowed_with(escalus_bosh, C).
+
+self_signed_cert_is_allowed_with(EscalusTransport, C) ->
+    UserSpec = generate_user(C, "alice-self-signed", EscalusTransport),
```

Comment:
Because here we are testing that the self-signed cert is allowed. This means that if the connection crashes (cert was not accepted) the test case crashes and we know sth went wrong. In the other place we expect the connection to crash. We spawn the process to handle in the same way 3 different transport, they are behaving a little bit differently and new process dying on connection failure was the common approach for every transport method.